net.sourceforge.mcomix.yml

```diff
@@ -0,0 +1,96 @@
+app-id: net.sourceforge.mcomix
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+rename-icon: mcomix
+rename-desktop-file: mcomix.desktop
+rename-appdata-file: mcomix.appdata.xml
+command: mcomix
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --device=dri
+  # need, for own file chooser
+  - --filesystem=host:ro
+
+modules:
+  - name: 7-zip
+    buildsystem: simple
+    build-commands:
+      - |
+        for bundle in Alone2 SFXCon; do
+          pushd CPP/7zip/Bundles/${bundle}
+          make -j ${FLATPAK_BUILDER_N_JOBS} -f ../../cmpl_gcc.mak
+          popd
+        done
+      - install -Dm755 CPP/7zip/Bundles/Alone2/b/g/7zz   $FLATPAK_DEST/bin/7z
+      - install -Dm755 CPP/7zip/Bundles/SFXCon/b/g/7zCon $FLATPAK_DEST/bin/7zCon.sfx
+
+    sources:
+      - type: archive
+        strip-components: 0
+        url: https://7-zip.org/a/7z2201-src.tar.xz
+        sha256: 393098730c70042392af808917e765945dc2437dee7aae3cfcc4966eb920fbc5
+
+  - name: unrar
+    no-autogen: true
+    build-options:
+      strip: true
+    make-install-args:
+      - DESTDIR=$(FLATPAK_DEST)
+    sources:
+      - type: archive
+        url: "https://www.rarlab.com/rar/unrarsrc-5.8.3.tar.gz"
+        sha256: 3591685c8f5bbcb0be09de3d0a0544adb88966b9cccb80986f6cd2b534fd91a6
+
+  - name: mcomix
+    buildsystem: simple
+    build-commands:
+      - python ./setup.py install --single-version-externally-managed --root '/' --prefix $FLATPAK_DEST 
+    sources:
+      - type: archive
+        url: https://downloads.sourceforge.net/project/mcomix/MComix-2.0.2/mcomix-2.0.2.tar.gz
+        sha256: ef38d0713e56a07c72f98cc20c9eacda780e39f3b82fd7b1baa06a69c79c0a58
+
+      - type: patch
+        path: appdata.patch
+
+    modules:
+      - name: pygobject
```

Comment:
why not use the GNOME runtime instead? It has all that is needed, doesn't it?

Reply:
I think freedesktop is enough, only needs py binding dependencies which are pretty small.